main.cpp

```diff
@@ -110,19 +99,20 @@ int main()
     T valchar2(6 , "M");                                             //6
     
     Compare f;                                            //7
-    auto* smaller = f.compare(&valchar1 ,&valchar2);                              //8
+    auto* smaller = f.compare(valchar1 ,valchar2);                              //8
     if (smaller != nullptr)
     {
         std::cout << "the smaller one is << " << smaller->name << std::endl; //9
     }
     else
     {
-        std::cout << "Null Pointer returned. Change inputs" <<std::endl;
+        std::cout << "Null pointer, choose another input" << std::endl;
```

Comment:
Now that the `compare` function takes references, there is only one reason for it to return `nullptr`.  Update this `std::cout` expression to specify why nullptr was returned.
- [ ] revise